app/web/src/components/AssetDetailIntrinsicInput.vue

```diff
@@ -157,13 +125,53 @@ const unsetFuncId = computed(() => {
   return func?.funcId as FuncId;
 });
 
+const initialFilter = (): DropdownFilter | null => {
+  if (display.value?.value?.startsWith("s_")) {
+    return "inputSocket";
+  } else if (display.value?.value?.startsWith("p_")) {
+    return "prop";
+  } else if (display.value?.funcId === unsetFuncId.value) {
+    return "unset";
+  } else if ("socketName" in props.data) {
+    // NOTE(nick): fallback to the input data if the display is empty. We need this in case a "emit" blows the
+    // component away. This could be cleaner to avoid having multiple branches.
+    return "inputSocket";
+  } else if ("path" in props.data) {
+    // NOTE(nick): fallback to the input data if the display is empty. We need this in case a "emit" blows the
+    // component away. This could be cleaner to avoid having multiple branches.
+    return "prop";
+  }
+  return null;
+};
+const selectedFilter = ref<DropdownFilter | null>(initialFilter());
+const selectFilter = (item: DropdownFilter) => {
+  if (props.isLocked) return;
+
+  selectedFilter.value = item;
+
+  if (item === "unset") {
+    emit("changeToUnset", display.value);
+  } else {
+    emit("changeToIdentity", display.value, null);
+  }
+};
```

Comment:
Wrapper around the old select to handle both the filter changing and the emit side effect.